internal/services/vpc/routes_data_source.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/scaleway/scaleway-sdk-go/api/vpc/v2"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/regional"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/types"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/verify"
+)
+
+func DataSourceRoutes() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: DataSourceRoutesRead,
+		Schema: map[string]*schema.Schema{
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Only routes within this VPC will be returned",
+			},
+			"nexthop_resource_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Only routes with a matching next hop resource ID will be returned",
+			},
+			"nexthop_private_network_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Only routes with a matching next hop private network ID will be returned",
+			},
+			"nexthop_resource_type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "Only Routes with a matching next hop resource type will be returned",
+				ValidateDiagFunc: verify.ValidateEnum[vpc.RouteWithNexthopResourceType](),
+			},
+			"is_ipv6": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Only routes with an IPv6 destination will be returned",
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "Routes with these exact tags are listed.",
+			},
+			"routes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"vpc_id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"tags": {
+							Computed: true,
+							Type:     schema.TypeList,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"created_at": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"description": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"destination": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"nexthop_resource_id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"nexthop_private_network_id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"nexthop_ip": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"nexthop_name": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"nexthop_resource_type": {
+							Computed: true,
+							Type:     schema.TypeString,
```

Comment:
Could you add documentation, for instance about the kind of value that could be shown off here